clickhouse-http-client/src/main/java/com/clickhouse/client/http/ClickHouseHttpConnection.java

```diff
@@ -417,10 +410,23 @@ protected String getDefaultUserAgent() {
         return config.getClientName();
     }
 
+    protected String getAdditionalFrameworkUserAgent() {
+        List<String> frameworks = List.of("apache.spark");
```

Comment:
This method will be called every time we do a request, `frameworks` is a static information. Please move it out of the method.